src/ServiceControl.AcceptanceTests/AcceptanceTest.cs

```diff
@@ -419,113 +288,358 @@ protected byte[] DownloadData(string url)
                 condition = _ => true;
             }
 
-            var response = Get<List<T>>(url);
-
+            var response = Get<List<T>>(url).GetAwaiter().GetResult();
+            item = null;
             if (response != null)
             {
                 var items = response.Where(i => condition(i)).ToList();
 
-                if (items.Count() > 1)
+                if (items.Count > 1)
                 {
                     throw new InvalidOperationException("More than one matching element found");
                 }
 
                 item = items.SingleOrDefault();
             }
-            else
+
+            if (item != null)
             {
-                item = null;
+                return true;
             }
 
-            if (item == null)
+            Task.Delay(1000).GetAwaiter().GetResult();
+
+            return false;
+        }
+
+        protected void Post<T>(string url, T payload = null) where T : class
+        {
+            if (!url.StartsWith("http://"))
             {
-                Thread.Sleep(1000);
+                url = $"http://localhost:{port}{url}";
+            }
 
-                return false;
+            var json = JsonConvert.SerializeObject(payload, serializerSettings);
+            var response = httpClient.PostAsync(url, new StringContent(json, null, "application/json")).GetAwaiter().GetResult();
+
+            Console.WriteLine($"{response.RequestMessage.Method} - {url} - {(int) response.StatusCode}");
+
+            if (!response.IsSuccessStatusCode)
+            {
+                var body = response.Content.ReadAsStringAsync().GetAwaiter().GetResult();
+                throw new InvalidOperationException($"Call failed: {(int) response.StatusCode} - {response.ReasonPhrase} - {body}");
             }
+        }
 
-            return true;
+        protected byte[] DownloadData(string url, HttpStatusCode successCode = HttpStatusCode.OK)
+        {
+            if (!url.StartsWith("http://"))
+            {
+                url = $"http://localhost:{port}/api{url}";
```

Comment:
This one is inconsistent with the logic of all the other methods (it adds the /api). Should these be consistent? Or even pulled out as a private helper method?